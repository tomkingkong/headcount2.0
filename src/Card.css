.Card {
  cursor: pointer;
  background-color: #fff;
  border-radius: 0.2rem;
  box-shadow: 0.1rem 0.1rem lightgrey;
  overflow: scroll;
  min-width: 8rem;
  width: 100%;
  max-width: 10rem;
  margin: 0.5rem 0.5rem;
  padding: 0.5rem;
  transition: all 0.25s linear;
}

.Card:hover {
  box-shadow: 0.4rem 0.4rem lightblue;
}

.Card ul {
  list-style-type: none;
  padding: auto;
}

.Card h1 {
  border-bottom: 0.2rem double transparent;
  font-size: 0.9rem;
  padding-bottom: 0.5rem;
  height: 1.8rem;
  text-align: center;
  transition: all 0.35s linear;
}

.Card:hover h1 {
  border-bottom: 0.2rem double lightblue;
  color: rgba(8, 107, 221);
}

.selected {
  border: 0.2rem solid rgb(18, 100, 194);
}

.Card .Green {
  color: rgb(12, 173, 12);
}

.Card .Red {
  color: rgb(243, 14, 14);
}