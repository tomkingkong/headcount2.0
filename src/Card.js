import React from 'react';

import './Card.css';

const Card = ({stats, location}) => {
  const dates = Object.keys(stats).map(year => <li className={stats[year]>0.5 ? 'Green' : 'Red'} key={location+year}>{year}: {stats[year]}</li>)
  
  return (
    <article className="Card">
      <h1>{location}</h1>
      <ul>{dates}</ul>
    </article>
  );
}

export default Card;