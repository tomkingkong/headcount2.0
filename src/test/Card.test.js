import React from 'react';
import { shallow } from 'enzyme';

import Card from '../Card';
const school = {
  'stats': { 
    2005: 0.521, 
    2017: 0.521, 
    2018: 0.521, 
    2019: 0.521 
  }, 
  'location': 'SHMOOP', 
  'id':'SHMOOP'
};

describe('Card', () => {
    
  it('should match snapshot with a data object', () => {
    const wrapper = shallow(<Card {...school} key={school.id}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot with a data object and function', () => {
    const selectSchool = jest.fn();
    const wrapper = shallow(<Card {...school} key={school.id} selected={selectSchool}/>);
    expect(wrapper).toMatchSnapshot();
  });


  beforeEach(() => {
    wrapper = shallow(<Card {...card} key={card.id}/>);
  });
  
  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

})