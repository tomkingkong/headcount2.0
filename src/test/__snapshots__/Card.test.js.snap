// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot with a data object 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    id="SHMOOP"
    location="SHMOOP"
    stats={
        Object {
            "2005": 0.521,
            "2017": 0.521,
            "2018": 0.521,
            "2019": 0.521,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <h1>
                SHMOOP
        </h1>
        <ul>
                <li
                        className="Green"
                >
                        2005
                        : 
                        0.521
                </li>
                <li
                        className="Green"
                >
                        2017
                        : 
                        0.521
                </li>
                <li
                        className="Green"
                >
                        2018
                        : 
                        0.521
                </li>
                <li
                        className="Green"
                >
                        2019
                        : 
                        0.521
                </li>
        </ul>
</div>,
      "className": "Card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            SHMOOP
</h1>,
          <ul>
            <li
                        className="Green"
            >
                        2005
                        : 
                        0.521
            </li>
            <li
                        className="Green"
            >
                        2017
                        : 
                        0.521
            </li>
            <li
                        className="Green"
            >
                        2018
                        : 
                        0.521
            </li>
            <li
                        className="Green"
            >
                        2019
                        : 
                        0.521
            </li>
</ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SHMOOP",
          },
          "ref": null,
          "rendered": "SHMOOP",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="Green"
>
                2005
                : 
                0.521
</li>,
              <li
                className="Green"
>
                2017
                : 
                0.521
</li>,
              <li
                className="Green"
>
                2018
                : 
                0.521
</li>,
              <li
                className="Green"
>
                2019
                : 
                0.521
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "SHMOOP2005",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2005",
                  ": ",
                  0.521,
                ],
                "className": "Green",
              },
              "ref": null,
              "rendered": Array [
                "2005",
                ": ",
                0.521,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "SHMOOP2017",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2017",
                  ": ",
                  0.521,
                ],
                "className": "Green",
              },
              "ref": null,
              "rendered": Array [
                "2017",
                ": ",
                0.521,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "SHMOOP2018",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2018",
                  ": ",
                  0.521,
                ],
                "className": "Green",
              },
              "ref": null,
              "rendered": Array [
                "2018",
                ": ",
                0.521,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "SHMOOP2019",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2019",
                  ": ",
                  0.521,
                ],
                "className": "Green",
              },
              "ref": null,
              "rendered": Array [
                "2019",
                ": ",
                0.521,
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <h1>
                    SHMOOP
          </h1>
          <ul>
                    <li
                              className="Green"
                    >
                              2005
                              : 
                              0.521
                    </li>
                    <li
                              className="Green"
                    >
                              2017
                              : 
                              0.521
                    </li>
                    <li
                              className="Green"
                    >
                              2018
                              : 
                              0.521
                    </li>
                    <li
                              className="Green"
                    >
                              2019
                              : 
                              0.521
                    </li>
          </ul>
</div>,
        "className": "Card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              SHMOOP
</h1>,
            <ul>
              <li
                            className="Green"
              >
                            2005
                            : 
                            0.521
              </li>
              <li
                            className="Green"
              >
                            2017
                            : 
                            0.521
              </li>
              <li
                            className="Green"
              >
                            2018
                            : 
                            0.521
              </li>
              <li
                            className="Green"
              >
                            2019
                            : 
                            0.521
              </li>
</ul>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SHMOOP",
            },
            "ref": null,
            "rendered": "SHMOOP",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="Green"
>
                  2005
                  : 
                  0.521
</li>,
                <li
                  className="Green"
>
                  2017
                  : 
                  0.521
</li>,
                <li
                  className="Green"
>
                  2018
                  : 
                  0.521
</li>,
                <li
                  className="Green"
>
                  2019
                  : 
                  0.521
</li>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "SHMOOP2005",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "2005",
                    ": ",
                    0.521,
                  ],
                  "className": "Green",
                },
                "ref": null,
                "rendered": Array [
                  "2005",
                  ": ",
                  0.521,
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "SHMOOP2017",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "2017",
                    ": ",
                    0.521,
                  ],
                  "className": "Green",
                },
                "ref": null,
                "rendered": Array [
                  "2017",
                  ": ",
                  0.521,
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "SHMOOP2018",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "2018",
                    ": ",
                    0.521,
                  ],
                  "className": "Green",
                },
                "ref": null,
                "rendered": Array [
                  "2018",
                  ": ",
                  0.521,
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "SHMOOP2019",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "2019",
                    ": ",
                    0.521,
                  ],
                  "className": "Green",
                },
                "ref": null,
                "rendered": Array [
                  "2019",
                  ": ",
                  0.521,
                ],
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Card should match snapshot with a data object and function 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    id="SHMOOP"
    location="SHMOOP"
    selected={[Function]}
    stats={
        Object {
            "2005": 0.521,
            "2017": 0.521,
            "2018": 0.521,
            "2019": 0.521,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <h1>
                SHMOOP
        </h1>
        <ul>
                <li
                        className="Green"
                >
                        2005
                        : 
                        0.521
                </li>
                <li
                        className="Green"
                >
                        2017
                        : 
                        0.521
                </li>
                <li
                        className="Green"
                >
                        2018
                        : 
                        0.521
                </li>
                <li
                        className="Green"
                >
                        2019
                        : 
                        0.521
                </li>
        </ul>
</div>,
      "className": "Card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            SHMOOP
</h1>,
          <ul>
            <li
                        className="Green"
            >
                        2005
                        : 
                        0.521
            </li>
            <li
                        className="Green"
            >
                        2017
                        : 
                        0.521
            </li>
            <li
                        className="Green"
            >
                        2018
                        : 
                        0.521
            </li>
            <li
                        className="Green"
            >
                        2019
                        : 
                        0.521
            </li>
</ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SHMOOP",
          },
          "ref": null,
          "rendered": "SHMOOP",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="Green"
>
                2005
                : 
                0.521
</li>,
              <li
                className="Green"
>
                2017
                : 
                0.521
</li>,
              <li
                className="Green"
>
                2018
                : 
                0.521
</li>,
              <li
                className="Green"
>
                2019
                : 
                0.521
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "SHMOOP2005",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2005",
                  ": ",
                  0.521,
                ],
                "className": "Green",
              },
              "ref": null,
              "rendered": Array [
                "2005",
                ": ",
                0.521,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "SHMOOP2017",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2017",
                  ": ",
                  0.521,
                ],
                "className": "Green",
              },
              "ref": null,
              "rendered": Array [
                "2017",
                ": ",
                0.521,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "SHMOOP2018",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2018",
                  ": ",
                  0.521,
                ],
                "className": "Green",
              },
              "ref": null,
              "rendered": Array [
                "2018",
                ": ",
                0.521,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "SHMOOP2019",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2019",
                  ": ",
                  0.521,
                ],
                "className": "Green",
              },
              "ref": null,
              "rendered": Array [
                "2019",
                ": ",
                0.521,
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <h1>
                    SHMOOP
          </h1>
          <ul>
                    <li
                              className="Green"
                    >
                              2005
                              : 
                              0.521
                    </li>
                    <li
                              className="Green"
                    >
                              2017
                              : 
                              0.521
                    </li>
                    <li
                              className="Green"
                    >
                              2018
                              : 
                              0.521
                    </li>
                    <li
                              className="Green"
                    >
                              2019
                              : 
                              0.521
                    </li>
          </ul>
</div>,
        "className": "Card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              SHMOOP
</h1>,
            <ul>
              <li
                            className="Green"
              >
                            2005
                            : 
                            0.521
              </li>
              <li
                            className="Green"
              >
                            2017
                            : 
                            0.521
              </li>
              <li
                            className="Green"
              >
                            2018
                            : 
                            0.521
              </li>
              <li
                            className="Green"
              >
                            2019
                            : 
                            0.521
              </li>
</ul>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SHMOOP",
            },
            "ref": null,
            "rendered": "SHMOOP",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="Green"
>
                  2005
                  : 
                  0.521
</li>,
                <li
                  className="Green"
>
                  2017
                  : 
                  0.521
</li>,
                <li
                  className="Green"
>
                  2018
                  : 
                  0.521
</li>,
                <li
                  className="Green"
>
                  2019
                  : 
                  0.521
</li>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "SHMOOP2005",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "2005",
                    ": ",
                    0.521,
                  ],
                  "className": "Green",
                },
                "ref": null,
                "rendered": Array [
                  "2005",
                  ": ",
                  0.521,
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "SHMOOP2017",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "2017",
                    ": ",
                    0.521,
                  ],
                  "className": "Green",
                },
                "ref": null,
                "rendered": Array [
                  "2017",
                  ": ",
                  0.521,
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "SHMOOP2018",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "2018",
                    ": ",
                    0.521,
                  ],
                  "className": "Green",
                },
                "ref": null,
                "rendered": Array [
                  "2018",
                  ": ",
                  0.521,
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "SHMOOP2019",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "2019",
                    ": ",
                    0.521,
                  ],
                  "className": "Green",
                },
                "ref": null,
                "rendered": Array [
                  "2019",
                  ": ",
                  0.521,
                ],
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Card should match snapshot with an empty data object 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    id=""
    location=""
    stats={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <h1 />
        <ul>
                
        </ul>
</div>,
      "className": "Card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1 />,
          <ul>
            
</ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": "ul",
        },
      ],
      "type": "div",
    },
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <h1 />
          <ul>
                    
          </ul>
</div>,
        "className": "Card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1 />,
            <ul>
              
</ul>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
            },
            "ref": null,
            "rendered": Array [],
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
